python/lsst/sims/catUtils/mixins/sncat.py

```diff
@@ -0,0 +1,302 @@
+    # 'mag_u', 'mag_g', 'mag_r', 'mag_i', 'mag_z', 'mag_y']
+
+    # Flux variables are convenient to display in exponential format to avoid
+    # having them cut off
+    variables = ['flux_u', 'flux_g', 'flux_r', 'flux_i', 'flux_z', 'flux_y']
+    variables += ['flux', 'flux_err', 'mag_err']
+
+    override_formats = {'snra': '%8e', 'sndec': '%8e', 'c': '%8e',
+                        'x0': '%8e'}
+    for var in variables:
+        override_formats[var] = '%8e'
+
+    cannot_be_null = ['x0', 'z', 't0']
+
+    @astropy.utils.lazyproperty
+    def mjdobs(self):
+        '''
+        The time of observation for the catalog, which is set to be equal
+        to obs_metadata.mjd
+        '''
+        return self.obs_metadata.mjd
+
+    @astropy.utils.lazyproperty
+    def badvalues(self):
+        '''
+        The representation of bad values in this catalog is numpy.nan
+        '''
+        return np.nan
+
+    @property
+    def suppressDimSN(self):
+        '''
+        Boolean to decide whether to output observations of SN that are too dim
+        should be represented in the catalog or not. By default set to True
+        '''
+        # self._suppressDimSN = True
+        if not hasattr(self, '_suppressDimSN'):
+            suppressDimSN_default = True
+            self._suppressDimSN = suppressDimSN_default
+        return self._suppressDimSN
+
+    @suppressDimSN.setter
+    def suppressDimSN(self, suppressDimSN):
+        '''
+        set the value of suppressDimSN of the catalog
+
+        Parameters
+        ----------
+        value : Boolean, mandatory
+            Value to set suppressDimSN to
+        '''
+        # if suppressDimSN is None:
+        #    self._suppressDimSN = True
+        # else:
+        self._suppressDimSN = suppressDimSN
+        return self._suppressDimSN
+
+    @property
+    def suppressHighzSN(self):
+        '''
+        Boolean to decide whether to output information corresponding to SN
+        at redshift above self.maxz
+        '''
+
+        return True
+
+    @property
+    def maxTimeSNVisible(self):
+        '''
+        The catalog will provide values for SN flux (even if zero according to
+        model) for peak \pm maxTimeVisibilityforSN
+        '''
+        return 100.
+
+    @property
+    def maxz(self):
+        return 1.2
+
+    @astropy.utils.lazyproperty
+    def photometricparameters(self, expTime=15., nexp=2):
+        lsstPhotometricParameters = PhotometricParameters(exptime=expTime,
+                                                          nexp=nexp)
+        return lsstPhotometricParameters
+
+    @astropy.utils.lazyproperty
+    def lsstBandpassDict(self):
+        return BandpassDict.loadTotalBandpassesFromFiles()
+
+    @astropy.utils.lazyproperty
+    def observedIndices(self):
+        bandPassNames = self.obs_metadata.bandpass
+        return [self.lsstBandpassDict.keys().index(x) for x in bandPassNames]
+
+    @astropy.utils.lazyproperty
+    def lsstpbase(self):
```

Comment:
It does not look like you use this property.  If so, I think it should be dropped.